bitcoin/src/taproot.rs

```diff
@@ -49,19 +49,23 @@ const MIDSTATE_TAPSIGHASH: [u8; 32] = [
 // Taproot test vectors from BIP-341 state the hashes without any reversing
 #[rustfmt::skip]
 sha256t_hash_newtype!(TapLeafHash, TapLeafTag, MIDSTATE_TAPLEAF, 64,
-    doc="Taproot-tagged hash for tapscript Merkle tree leafs", false
+    doc="Taproot-tagged hash with tag \"TapLeaf\". This is used for computing tapscript
+    script spend hash.", false
 );
 #[rustfmt::skip]
-sha256t_hash_newtype!(TapBranchHash, TapBranchTag, MIDSTATE_TAPBRANCH, 64,
-    doc="Taproot-tagged hash for tapscript Merkle tree branches", false
+sha256t_hash_newtype!(TapNodeHash, TapBranchTag, MIDSTATE_TAPBRANCH, 64,
+    doc="Taproot-tagged hash with tag \"TapBranch\". This hash type is used while constructing
+    a taproot tree.", false
```

Comment:
In 6609dae247e9ac63c2a1d357183c93a55e4fb1af:

This doccomment should be updated to say that it might be tagged either with `TapBranch` or with `TapLeaf`; or maybe it should not mention this at all and just say "tagged hash used in taproot trees; see BIP-340 for tagging rules"